ArduPlane/system.cpp

```diff
@@ -144,14 +144,10 @@ void Plane::init_ardupilot()
     usb_connected = true;
     check_usb_mux();
 
-    // setup serial port for telem1
-    gcs[1].setup_uart(serial_manager, AP_SerialManager::SerialProtocol_MAVLink, 0);
-
-    // setup serial port for telem2
-    gcs[2].setup_uart(serial_manager, AP_SerialManager::SerialProtocol_MAVLink, 1);
-
-    // setup serial port for fourth telemetry port (not used by default)
-    gcs[3].setup_uart(serial_manager, AP_SerialManager::SerialProtocol_MAVLink, 2);
+    // setup all other telem slots with  serial ports
+    for(int i = 1; i < MAVLINK_COMM_NUM_BUFFERS; i++) {
```

Comment:
Both easily addressable, will fix up and push it in a bit.

On Mon, Mar 21, 2016 at 4:50 PM, Lucas De Marchi <notifications@github.com>
wrote:

> In ArduPlane/system.cpp
> <https://github.com/diydrones/ardupilot/pull/3752#discussion_r56917360>:
>
> > @@ -144,14 +144,10 @@ void Plane::init_ardupilot()
> >      usb_connected = true;
> >      check_usb_mux();
> >
> > -    // setup serial port for telem1
> > -    gcs[1].setup_uart(serial_manager, AP_SerialManager::SerialProtocol_MAVLink, 0);
> > -
> > -    // setup serial port for telem2
> > -    gcs[2].setup_uart(serial_manager, AP_SerialManager::SerialProtocol_MAVLink, 1);
> > -
> > -    // setup serial port for fourth telemetry port (not used by default)
> > -    gcs[3].setup_uart(serial_manager, AP_SerialManager::SerialProtocol_MAVLink, 2);
> > +    // setup all other telem slots with  serial ports
> > +    for(int i = 1; i < MAVLINK_COMM_NUM_BUFFERS; i++) {
>
> and some would say to use uint8_t, too....
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/diydrones/ardupilot/pull/3752/files/cf3ce746b73a5321e94899efc8e82b1922fcf3cd#r56917360>
>